superdesk/text_checkers/spellcheckers/leuven_dutch.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013-2019 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+import os
+import logging
+import requests
+from superdesk.errors import SuperdeskApiError
+from superdesk.text_checkers.spellcheckers.base import SpellcheckerBase
+
+logger = logging.getLogger(__name__)
+OPT_API_KEY = "LEUVEN_DUTCH_API_KEY"
+API_URL = "https://dev.schrijfhulp.be/v1/{method}/"
+START_MARKER = "##__ERR_START__##"
+END_MARKER = "##__ERR_END__##"
+
+
+class LeuvenDutch(SpellcheckerBase):
+    """University of Leuven Dutch grammar/spellchecker integration
+
+    The LEUVEN_DUTCH_API_KEY setting (or environment variable) must be set to the API key
+    """
+
+    name = "leuven_dutch"
+    label = "University of Leuven Dutch spellchecker"
+    capacities = ["spelling"]
+    languages = ['nl']
+
+    def __init__(self, app):
+        super().__init__(app)
+        self.api_key = self.config.get(OPT_API_KEY, os.environ.get(OPT_API_KEY))
+
+    def check(self, text):
+        check_url = API_URL.format(method="spellingchecker")
+        data = {
+            "key": self.api_key,
+            "input": text,
+            "startmarker": START_MARKER,
+            "endmarker": END_MARKER,
+        }
+        r = requests.post(check_url, data=data)
+        if r.status_code != 200:
+            raise SuperdeskApiError.internalError("Unexpected return code from {}".format(self.name))
+
+        data = r.json()
+
+        err_list = []
+        check_data = {'errors': err_list}
+        for mistake in data['spellingchecker']['output'].get('mistakes', []):
+            idx = text.find(mistake)
```

Comment:
what about case when you have same mistake multiple times in text?

Reply:
only the first one will be underlined. I've actually thought about this one too, and I was thinking it was an acceptable compromise as it's an edge case and their output format is not that nice (marking and mistakes are not directly linked).

But after thinking about it again it should not be such a problem to parse and associate to spelling mistake, and it's more clean indeed, so I'll update the PR to work with the marked text.

Reply:
@petrjasek I've changed the code it's not using `mistakes` anymore, only the marker. Now the same mistake done several times is underlined every time as expected.